File: KevinBankersen/ql/src/main/java/eu/bankersen/kevin/ql/ast/expr/math/Pos.java
package eu.bankersen.kevin.ql.ast.expr.math;

import java.math.BigDecimal;

import eu.bankersen.kevin.ql.ast.BasicVisitor;
import eu.bankersen.kevin.ql.ast.expr.EvaluateExeption;
import eu.bankersen.kevin.ql.ast.expr.Expr;
import eu.bankersen.kevin.ql.ast.expr.NumberExpr;
import eu.bankersen.kevin.ql.typechecker.symboltable.SymbolTable;

public class Pos extends NumberExpr {
    
    public Pos(final Expr expr, final int line) {
	super(expr, null, line);
    }

    @Override
    public final BigDecimal evalExpr(SymbolTable symbolTable) throws EvaluateExeption {
	return ((BigDecimal) lhs().evalExpr(symbolTable)).abs();
    }
    
    @Override
    public <T> void accept(BasicVisitor v, T context) {
	v.visit(this);
    }
   
}

File: Edward Poot/src/ql/ast/expr/unaryoperatorexpr/pos.go
package unaryoperatorexpr

import (
	"math"
	"ql/ast/expr"
	"ql/ast/visit"
)

type Pos struct {
	Value expr.Expr
}

func (p Pos) GetValue() expr.Expr {
	return p.Value
}

func (p Pos) Eval(s interface{}) interface{} {
	return int(math.Abs(float64(p.GetValue().Eval(s).(int))))
}

func (p Pos) Accept(v visit.Visitor, s interface{}) interface{} {
	return v.Visit(p, s)
}